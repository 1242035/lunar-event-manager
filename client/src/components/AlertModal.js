import React from 'react';
import classNames from 'classnames';

/**
 * Modal that shows an error to the user.
 * @param{props} - passed properties
 * @return{React.Component}
 */
const AlertModal = (props) => {
  const {
    message, show, type,
  } = props.alertModalState;

  const classes = classNames({ 'io-modal': true, hide: !show });

  let header;
  if (type === 'error') {
    header = 'ERROR!';
  } else {
    header = 'SUCCESS!';
  }

  return (
    <div className={`${classes} ${type} io-error`} onClick={props.hideAlertModal}>
      <div className="io-modal-body io-error">
        <div className="io-header io-error">{header}</div>
        <div className="io-body">
          <div className="io-content io-error">
            <h2>{message} </h2>
          </div>
        </div>
        <div className="io-footer io-error">
          <button id="view-center-ok" className="io-submit-btn io-sm" onClick={props.hideAlertModal}>OK </button>
        </div>
      </div>
    </div>
  );
};

export default AlertModal;
