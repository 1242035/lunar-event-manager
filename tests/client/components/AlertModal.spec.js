import React from 'react';
import { shallow } from 'enzyme';
import AlertModal from '../../../client/src/components/AlertModal';

describe('AlertModal', () => {
  let wrapper;
  const alertModalState = {
    message: '',
    show: true,
    type: 'error',
  };

  it('Should maintain existing snapshot', () => {
    wrapper = shallow(<AlertModal alertModalState={alertModalState} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render connected component', () => {
    wrapper = shallow(<AlertModal alertModalState={alertModalState} />);
    expect(wrapper.length).toEqual(1);
  });
});
