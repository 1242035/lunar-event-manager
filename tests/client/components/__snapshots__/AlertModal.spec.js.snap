// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertModal Should maintain existing snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AlertModal
    alertModalState={
      Object {
        "message": "",
        "show": true,
        "type": "error",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="io-modal-body io-error"
      >
        <div
          className="io-header io-error"
        >
          ERROR!
        </div>
        <div
          className="io-body"
        >
          <div
            className="io-content io-error"
          >
            <h2>
              
               
            </h2>
          </div>
        </div>
        <div
          className="io-footer io-error"
        >
          <button
            className="io-submit-btn io-sm"
            id="view-center-ok"
          >
            OK 
          </button>
        </div>
      </div>,
      "className": "io-modal error io-error",
      "onClick": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="io-header io-error"
          >
            ERROR!
          </div>,
          <div
            className="io-body"
          >
            <div
              className="io-content io-error"
            >
              <h2>
                
                 
              </h2>
            </div>
          </div>,
          <div
            className="io-footer io-error"
          >
            <button
              className="io-submit-btn io-sm"
              id="view-center-ok"
            >
              OK 
            </button>
          </div>,
        ],
        "className": "io-modal-body io-error",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "ERROR!",
            "className": "io-header io-error",
          },
          "ref": null,
          "rendered": "ERROR!",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="io-content io-error"
            >
              <h2>
                
                 
              </h2>
            </div>,
            "className": "io-body",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h2>
                
                 
              </h2>,
              "className": "io-content io-error",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "",
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                "",
                " ",
              ],
              "type": "h2",
            },
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="io-submit-btn io-sm"
              id="view-center-ok"
            >
              OK 
            </button>,
            "className": "io-footer io-error",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "OK ",
              "className": "io-submit-btn io-sm",
              "id": "view-center-ok",
              "onClick": undefined,
            },
            "ref": null,
            "rendered": "OK ",
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="io-modal-body io-error"
        >
          <div
            className="io-header io-error"
          >
            ERROR!
          </div>
          <div
            className="io-body"
          >
            <div
              className="io-content io-error"
            >
              <h2>
                
                 
              </h2>
            </div>
          </div>
          <div
            className="io-footer io-error"
          >
            <button
              className="io-submit-btn io-sm"
              id="view-center-ok"
            >
              OK 
            </button>
          </div>
        </div>,
        "className": "io-modal error io-error",
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="io-header io-error"
            >
              ERROR!
            </div>,
            <div
              className="io-body"
            >
              <div
                className="io-content io-error"
              >
                <h2>
                  
                   
                </h2>
              </div>
            </div>,
            <div
              className="io-footer io-error"
            >
              <button
                className="io-submit-btn io-sm"
                id="view-center-ok"
              >
                OK 
              </button>
            </div>,
          ],
          "className": "io-modal-body io-error",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "ERROR!",
              "className": "io-header io-error",
            },
            "ref": null,
            "rendered": "ERROR!",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="io-content io-error"
              >
                <h2>
                  
                   
                </h2>
              </div>,
              "className": "io-body",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h2>
                  
                   
                </h2>,
                "className": "io-content io-error",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "",
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "",
                  " ",
                ],
                "type": "h2",
              },
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="io-submit-btn io-sm"
                id="view-center-ok"
              >
                OK 
              </button>,
              "className": "io-footer io-error",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "OK ",
                "className": "io-submit-btn io-sm",
                "id": "view-center-ok",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": "OK ",
              "type": "button",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
